test/test_stages.py

```diff
@@ -0,0 +1,36 @@
+import json
+import os
+
+from test import osbuildtest
+
+
+class TestDescriptions(osbuildtest.TestCase):
+
+    def run_stage_test(self, test_dir: str):
+        pipeline1 = f"{test_dir}/a.json"
+        pipeline2 = f"{test_dir}/b.json"
+        tree1, _ = self.run_osbuild(pipeline1)
+        tree2, _ = self.run_osbuild(pipeline2)
+
+        actual_diff = self.run_tree_diff(tree1, tree2)
+
+        with open(f"{test_dir}/diff.json") as f:
+            expected_diff = json.load(f)
+
+        self.assertTreeDiffsEqual(expected_diff, actual_diff)
+
+
+def generate_test_case(test):
+    def test_case(self):
+        self.run_stage_test(test)
+
+    return test_case
+
+
+def init_tests():
+    test_dir = 'test/stages_tests'
+    for test in os.listdir(test_dir):
+        setattr(TestDescriptions, f"test_{test}", generate_test_case(f"{test_dir}/{test}"))
```

Comment:
I prefer using `TestCase.subTest()` for parameterized tests. No need to change to that, just wanted to mention it.

Reply:
`setUp` and `tearDown` don't work with `subTest()`

Reply:
Ah! Seriously?! Thanks for pointing that out.

Reply:
I've tested this once again today's morning and yeah, it works this way. But I guess it's a correct behaviour, if setUp got called before each subTest, it would got called twice - once before test method and then before subtest, which would be imho really confusing.